src/mqt/benchviewer/backend.py

```diff
@@ -23,256 +23,544 @@
+
+            elif 0 < int(identifier) <= len(self.benchmarks) + len(self.nonscalable_benchmarks):
+                name = self.nonscalable_benchmarks[int(identifier) - 1 - len(self.benchmarks)]["filename"]
+                selected_nonscalable_benchmarks.append(name)
+
+        db_tmp = self.database.loc[
+            (
+                (self.database["num_qubits"] >= min_qubits)
+                & (self.database["num_qubits"] <= max_qubits)
+                & (self.database["benchmark"].isin(selected_scalable_benchmarks))
+            )
+            | (self.database["benchmark"].isin(selected_nonscalable_benchmarks))
+        ]
+
+        if indep_qiskit_compiler:
+            db_tmp1 = db_tmp.loc[(db_tmp["indep_flag"]) & (db_tmp["compiler"] == "qiskit")]
+            db_filtered = pd.concat([db_filtered, db_tmp1])
+
+        if indep_tket_compiler:
+            db_tmp2 = db_tmp.loc[(db_tmp["indep_flag"]) & (db_tmp["compiler"] == "tket")]
+            db_filtered = pd.concat([db_filtered, db_tmp2])
+
+        if nativegates_qiskit_compiler:
+            for gate_set in native_gatesets:
+                for opt_lvl in native_qiskit_opt_lvls:
+                    db_tmp3 = db_tmp.loc[
+                        (db_tmp["nativegates_flag"])
+                        & (db_tmp["gate_set"] == gate_set)
+                        & (db_tmp["compiler"] == "qiskit")
+                        & (db_tmp["compiler_settings"] == opt_lvl)
+                    ]
+                    db_filtered = pd.concat([db_filtered, db_tmp3])
+
+        if nativegates_tket_compiler:
+            for gate_set in native_gatesets:
+                db_tmp4 = db_tmp.loc[
+                    (db_tmp["nativegates_flag"]) & (db_tmp["gate_set"] == gate_set) & (db_tmp["compiler"] == "tket")
+                ]
+                db_filtered = pd.concat([db_filtered, db_tmp4])
+
+        if mapped_qiskit_compiler:
+            for opt_lvl in mapped_qiskit_opt_lvls:
+                for device in mapped_devices:
+                    db_tmp5 = db_tmp.loc[
+                        (db_tmp["mapped_flag"])
+                        & (db_tmp["target_device"] == device)
+                        & (db_tmp["compiler"] == "qiskit")
+                        & (db_tmp["compiler_settings"] == opt_lvl)
+                    ]
+                    db_filtered = pd.concat([db_filtered, db_tmp5])
+
+        if mapped_tket_compiler:
+            for placement in mapped_tket_placements:
+                for device in mapped_devices:
+                    db_tmp6 = db_tmp.loc[
+                        (db_tmp["mapped_flag"])
+                        & (db_tmp["target_device"] == device)
+                        & (db_tmp["compiler"] == "tket")
+                        & (db_tmp["compiler_settings"] == placement)
+                    ]
+                    db_filtered = pd.concat([db_filtered, db_tmp6])
+
+        return db_filtered["path"].to_list()
+
+    class NoSeekBytesIO:
```

Comment:
I am not quite sure I understand where this class is coming from. Could you elaborate?

Reply:
To be honest, this was what Stefan implemented to enable the streaming when download benchmarks from the MQT Bench website. I will check it out in more detail tomorrow.

Reply:
Ah, if that's the case, then this should be good. I just didn't see where this was coming from in the diff and thought it was new.